### PR TITLE
Added benchmark for SSR

### PR DESCRIPTION
**Pull request by blake-newman**
- Includes `renderToStream` and`renderToString`
- Added SSR benchmark README.md
- Renders table with 10000 rows with 100 columns (1 million cells)

### CODE REVIEW

**Comment by yyx990803** [comment]
Hold this (will need to amend after data observation skipping is properly implemented)

**Comment by blake-newman** [comment]
Yup, put up so others can inspect code in the mean time.

File: benchmarks/ssr/common.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  template: '<div><h1>{{ performance }}ms</h1><my-table></my-table></div>',
+  mixins: [perfMixin],
+  components: {
+    myTable: {
+      data: function () {
+        return {
+          grid: gridData
+        }
+      },
+      template: '<table width="100%" cellspacing="2"><row v-for="row in grid"></row></table>',
```
Comment (yyx990803) [inline benchmarks/ssr/common.js]: actually the benchmark is broken because you are not passing in the `:row` prop at all :/

Comment (blake-newman) [inline benchmarks/ssr/common.js]: I was wondering why all of a sudden node was able to render 10 million cells without running into memory issues.

Fixed.

File: benchmarks/ssr/common.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        return {
+          grid: gridData
+        }
+      },
+      template: '<table width="100%" cellspacing="2"><row v-for="row in grid" :row="row"></row></table>',
+      components: {
+        row: {
+          props: ['row'],
+          mixins: [perfMixin],
+          template: '<tr><th>{{ performance }}ms</th><column v-for="item in row"></column></tr>',
```
Comment (yyx990803) [inline benchmarks/ssr/common.js]: Here it should be `v-for="item in row.items"`... previous versions were ignoring the columnCount all together. This explains why numbers were off - we should probably just test 1000 x 10 which is a more practical scenario (Note it's 10k components, not just cells)

File: benchmarks/ssr/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+This benchmark renders a table of 10000 rows with 100 columns (1 million cells). It is extremely likely that you will never render 1 million table cells, this benchmark is to demonstrate the overall speeds of Vue.js SSR and time to content comparison between `renderToString` and `renderToStream`.
+
+To view the results follow the run section. Note that the overall completion time for the results are variable, this is due to other system related variants at run time (available memory, processing ect). In ideal circumstances both should finish within equal times.
+
+`renderToStream` pipes the content through a stream which gives massive performance benefits over renderToString. This can be observed through the benchmark.
+
+### run
+
+``` bash
+VUE_ENV=server node renderToString.js
```
Comment (yyx990803) [inline benchmarks/ssr/README.md]: Also add `NODE_ENV=production` here - probably make it an npm script to run both.

**Pull request merged**